crates/solver/src/driver.rs

```diff
@@ -178,18 +180,31 @@ impl Driver {
             .await
             .context("error retrieving current auction")?;
 
+        // It doesn't make sense to solve the same auction again because we wouldn't be able to
+        // store competition info etc.
+        if matches!(self.last_attempted_settlement, Some(id) if id == auction.id) {
+            tracing::debug!("skipping run because auction hasn't changed {}", auction.id);
+            return Ok(());
+        }
+
         let id = auction.id;
         let run = self.next_run_id();
 
         // extra function so that we can add span information
-        self.single_auction(auction, run)
+        let settlement_attempted = self
+            .single_auction(auction, run)
             .instrument(tracing::info_span!("auction", id, run))
             .await?;
+
+        if settlement_attempted {
+            self.last_attempted_settlement = Some(id);
+        }
+
         Ok(())
     }
 
-    /// Returns whether a settlement was attempted and thus a solver competition info stored.
-    async fn single_auction(&mut self, auction: AuctionWithId, run_id: u64) -> Result<()> {
+    /// Returns whether a settlement transaction was attempted.
+    async fn single_auction(&mut self, auction: AuctionWithId, run_id: u64) -> Result<bool> {
```

Comment:
Wouldn't it make sense to allow retrying the same auction if we submitted a tx but it got reverted?

Reply:
No because we cannot the per auction data again like competition info, order rewards.